tests/src/tracing/eventpipe/eventsvalidation/GCStartStop.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.Diagnostics.Tools.RuntimeClient;
+using Microsoft.Diagnostics.Tracing;
+using Tracing.Tests.Common;
+
+namespace Tracing.Tests.GCStartStop
+{
+    public sealed class MyEvents
+    {
+        private MyEvents() {}
+        public static MyEvents Log = new MyEvents();
+        public void MyEvent1() 
+        {
+            MyEvents myEvent = new MyEvents();
+            myEvent = null;
+            GC.Collect();
+        }
+    }
+
+    public class ProviderValidation
+    {
+        public static int Main(string[] args)
+        {
+            var providers = new List<Provider>()
+            {
+                new Provider("Microsoft-DotNETCore-SampleProfiler"),
+                new Provider("Microsoft-Windows-DotNETRuntime", 1, EventLevel.Informational)
+            };
+            
+            var configuration = new SessionConfiguration(circularBufferSizeMB: 1024, format: EventPipeSerializationFormat.NetTrace,  providers: providers);
+            return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, configuration, _DoesTraceContainGCEvents);
+        }
+
+        private static Dictionary<string, ExpectedEventCount> _expectedEventCounts = new Dictionary<string, ExpectedEventCount>()
+        {
+            { "Microsoft-Windows-DotNETRuntime", -1 },
+            { "Microsoft-Windows-DotNETRuntimeRundown", -1 },
+            { "Microsoft-DotNETCore-SampleProfiler", -1 }
+        };
+
+        private static Action _eventGeneratingAction = () => 
+        {
+            foreach (var _ in Enumerable.Range(0,1000))
+            {
+                MyEvents.Log.MyEvent1();
+            }
+        };
+
+        private static Func<EventPipeEventSource, Func<int>> _DoesTraceContainGCEvents = (source) => 
+        {
+            int GCStartEvents = 0;
+            int GCEndEvents = 0;
+            source.Clr.GCStart += (eventData) => GCStartEvents += 1;
+            source.Clr.GCStop += (eventData) => GCEndEvents += 1;
+            return () => GCStartEvents == 1000 && GCEndEvents == 1000 ? 100 : -1;
```

Comment:
I agree with @josalem here - Let's just make sure we also check that # of GCStart and # of GCEnd events are the same instead of simply changing the check above to `GCStartEvents > 1000 && GCEndEvents > 1000`.